osu.Game/Online/Multiplayer/StatefulMultiplayerClient.cs

```diff
@@ -84,7 +84,7 @@ protected StatefulMultiplayerClient()
             IsConnected.BindValueChanged(connected =>
             {
                 // clean up local room state on server disconnect.
-                if (!connected.NewValue)
+                if (!connected.NewValue && Room != null)
```

Comment:
Did this actually silence/fix anything? It looks like `LeaveRoom` should be handling the null case correctly already (albeit slightly awkwardly because inheritance) - maybe that logic should be trimmed if we're moving this up here?

Reply:
This should fix the error message showing up at the main menu.

Reply:
Ohhhhh right gotcha. Got totally fixated on the `LeaveRoom` operation while simultaneously glossing over the log statement that shows up within.